### ISSUE TITLE
Unwanted first character using IME API

### ISSUE THREAD

**Issue by budnix** [handsontable/handsontable#4662]
### Description

The unwanted first character appears in the editor when IME API is used.

### Steps to reproduce

1. Select Chinse (pinyin) keyboard layout;
2. Go to https://handsontable.com/features delete text `Task 33` (cell A1) and switch cell into editing mode (press <kbd>enter</kbd>);
3. Press <kbd>esc</kbd> to exit from editing mode;
4. Press letters <kbd>c</kbd><kbd>e</kbd><kbd>s</kbd>

In result, the editor should contain only Chinese symbols. Instead, the first character is visible as English character.

### Your environment
* Handsontable version: 0.35.0
* Browser Name and version: *
* Operating System: macOS

**Comment by budnix** [comment]
This https://github.com/handsontable/handsontable/issues/4661 issue is the same as this but with different behavior on Safari.

**Comment by krzysztofspilka** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Unwanted first character using IME API #4662](https://trello.com/c/1CWoALSh/92-unwanted-first-character-using-ime-api-4662)

**Comment by budnix** [comment]
The issue is rescheduled to February.

Tasks to be finished/fixed:
* The IME editor shouldn't appear on checkbox editor and on cells marked as `readOnly`;
* Add some tests to cover that editing is working for Asia users;
* On FF the bug with unwanted first character still exists but only limited to `dropdown` and `autocomplete` editors;
* Check if focus which is stolen by `copyPaste` plugin on selection end breaks the editing;

**Comment by ruzia** [comment]
any new update?
I'm waiting for fixed it.

**Comment by AMBudnik** [comment]
Hi @ruzia 

there are still a few bugs that we need to finish before we will be able to get back to this project. I do not know if we will be able to do in in February (as predicted before) but we will do our best to try.

**Comment by budnix** [comment]
The IME was merged into develop branch. The editor is not compatible with `fragmentSelection` set to `true`. As a side effect, the IME tooltip appears when checkbox and select editor is set and when `editor: false` is set or cells are `readOnly`.

**Comment by budnix** [comment]
Additionally, there is some visual bug which occurs only on Edge browser. When we try to compose the word the first character is not visible in the input element but psychically it is there so it does not disturb later composition. After inputting the second character and so on, the whole text becomes visible. This will be fixed as a separate issue.

**Comment by ruzia** [comment]
@budnix 

thanks for reply!
I'll check 4.0.0 later. 😄
